stages/org.osbuild.sysconfig

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python3
+"""
+
+Configure sysconfig files
+
+The sysconfig directory contains a variety of system configuration files. The
+values found in each of the configuration files may need to be specified
+before first boot.
+
+Currently, only the kernel and network configs can be modified.
+
+"""
+
+import sys
+import os
+
+import osbuild.api
+
+
+SCHEMA = """
+"additionalProperties": false,
+"properties": {
+  "kernel": {
+    "additionalProperties": false,
+    "type": "object",
+    "description": "sysconfig kernel options",
+    "properties": {
+      "update_default": {
+        "type": "boolean",
+        "description": "This option makes a newly installed kernel as the default in the boot entry selection."
+      },
+      "default_kernel": {
+        "type": "string",
+        "description": "This option specifies what package type will be used as the default."
+      }
+    }
+  },
+  "network": {
+    "additionalProperties": false,
+    "type": "object",
+    "description": "sysconfig network options",
+    "properties": {
+      "networking": {
+        "type": "boolean",
+        "description": "Enables or disables networking"
+      },
+      "no_zero_conf": {
+        "type": "boolean",
+        "description": "Disables the zero configuration network suite"
+      }
+    }
+  }
+}
+"""
+
+
+# sysconfig uses yes and no instead of true and false
+def bool_to_string(bool):
+    if bool:
+        return "yes"
+    else:
+        return "no"
+
+
+def configure_kernel(tree, kernel_options):
+    with open(f"{tree}/etc/sysconfig/kernel", 'w') as kernel_file:
+        for option, value in kernel_options.items():
+            if option == "update_default":
+                kernel_file.write(f"UPDATEDEFAULT={bool_to_string(value)}\n")
+            if option == "default_kernel":
+                kernel_file.write(f"DEFAULTKERNEL={value}\n")
+
+    return 0
+
+
+def configure_network(tree, network_options):
+    with open(f"{tree}/etc/sysconfig/network", 'w') as network_file:
+        for option, value in network_options.items():
+            if option == "networking":
+                network_file.write(f"NETWORKING={bool_to_string(value)}\n")
+            if option == "no_zero_conf":
+                network_file.write(f"NOZEROCONF={bool_to_string(value)}\n")
+
+    return 0
+
+
+def main(tree, options):
+    # if the sysconfig dir does not yet exist create it
```

Comment:
It belongs to the `filesystem` package so should really be there, but I guess it does not hurt, although I would tend to just do `os.makedirs(, exist_ok=True)` instead of checking and creating (which in /theory/ is racy).

Reply:
Done